### PR TITLE
KEP 2040: Kubelet CRI support

### PR DESCRIPTION
**Pull request by mrunalp**
Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by mrunalp** [comment]
@derekwaynecarr @dchen1107 ptal

File: keps/sig-node/2040-cri-beta/README.md
Comment (derekwaynecarr) [inline keps/sig-node/2040-cri-beta/README.md:2]: let's just use this KEP to identify changes from the alpha to beta and next release beta to stable.

So let's title this: "Kubelet CRI support", so it matches the tracking issue.
https://github.com/kubernetes/enhancements/issues/2040

File: keps/sig-node/2040-cri-beta/README.md
Comment (derekwaynecarr) [inline keps/sig-node/2040-cri-beta/README.md:36]: since CRI development has already happened, lets mark this with an [x]

File: keps/sig-node/2040-cri-beta/README.md
Comment (derekwaynecarr) [inline keps/sig-node/2040-cri-beta/README.md:52]: let's rephrase:

Identify remaining gaps to promote CRI to Beta and GA to reflect its practical use in production for many years.

File: keps/sig-node/2040-cri-beta/README.md
Comment (derekwaynecarr) [inline keps/sig-node/2040-cri-beta/README.md:68]: Evolve the CRI API version as we address feedback in each milestone towards stable.

- v1alpha2 (alpha, current state)
- v1beta (beta, proposed 1.20)
- v1 (stable, tbd)

File: keps/sig-node/2040-cri-beta/README.md
Comment (derekwaynecarr) [inline keps/sig-node/2040-cri-beta/README.md:71]: Let's just enumerate (so we can add anything else we discover between beta and GA) in subsequent release.

How about a table format like:

Risks
risk name, detail, mitigation plane

so:
metrics gathering, performance impact with cri and cadvisor stats gathering, measure impact with report to community.

etc.

File: keps/sig-node/2040-cri-beta/README.md
Comment (derekwaynecarr) [inline keps/sig-node/2040-cri-beta/README.md:79]: let's title this section:

changes between v1alpha1 and v1beta1

we can just add a new section then in 1.21 for v1beta1 to v1 while updating the same kep.

File: keps/sig-node/2040-cri-beta/README.md
Comment (derekwaynecarr) [inline keps/sig-node/2040-cri-beta/README.md:77]: Add a work item to no longer map the following flags as experimental:
container-runtime-endpoint

I think we should keep the image-service-endpoint experimental and evaluate if it makes sense or not to just remove it at the moment as a configurable flag.  I am curious if anyone is using the flags with separate values.

File: keps/sig-node/2040-cri-beta/kep.yaml
Comment (derekwaynecarr) [inline keps/sig-node/2040-cri-beta/kep.yaml:28]: 1.4

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
A few changes requested on formatting so we can use the same document for stable.

Other minor changes to discuss updates to the kubelet flags that are marked experimental.

This is lgtm from me after those updates.

File: keps/sig-node/2040-cri-beta/README.md
Comment (mrunalp) [reply keps/sig-node/2040-cri-beta/README.md:80]: Added.

File: keps/sig-node/2040-cri-beta/kep.yaml
Comment (mrunalp) [inline keps/sig-node/2040-cri-beta/kep.yaml:28]: Set to 1.5 per https://kubernetes.io/blog/2016/12/container-runtime-interface-cri-in-kubernetes/

File: keps/sig-node/2040-cri-beta/README.md
Comment (mrunalp) [reply keps/sig-node/2040-cri-beta/README.md]: done

File: keps/sig-node/2040-cri-beta/README.md
Comment (mrunalp) [reply keps/sig-node/2040-cri-beta/README.md]: Yeah, done :)

File: keps/sig-node/2040-cri-beta/README.md
Comment (mrunalp) [reply keps/sig-node/2040-cri-beta/README.md]: Done

**Comment by derekwaynecarr** [comment]
/approve
/lgtm

/hold
I want to make sure @dchen1107 has a moment to review as well.

File: keps/sig-node/2040-kubelet-cri/README.md
Comment (dchen1107) [inline keps/sig-node/2040-kubelet-cri/README.md:89]: I believe it is not deprecated by SIG network yet. cc/ @freehan from SIG network.

File: keps/sig-node/2040-kubelet-cri/README.md
Comment (dchen1107) [inline keps/sig-node/2040-kubelet-cri/README.md:76]: Thanks for including this as the risk here. But want to point out that promoting CRI to beta doesn't mean the node completely switch away from cadvisor for stats. They should be orthogonal. Meanwhile, we do need to know the performance status here as part of the beta process.

**Comment by dchen1107** [comment]
Added some comments, but nothing is blocking this KEP.

/lgtm
/approve

**Comment by derekwaynecarr** [comment]
thanks @dchen1107 
/hold cancel

File: keps/sig-node/2040-kubelet-cri/README.md
Comment (freehan) [reply keps/sig-node/2040-kubelet-cri/README.md:89]: Yes. Kubenet is not deprecated yet. @kubernetes/sig-network-misc

**Pull request merged**
